crypto/params.c

```diff
@@ -788,21 +788,22 @@ static int get_string_internal(const OSSL_PARAM *p, void **val, size_t max_len,
     if (used_len != NULL)
         *used_len = sz;
 
-    if (sz == 0)
-        return 1;
     if (p->data == NULL)
         return 0;
 
     if (val == NULL)
         return 1;
 
     if (*val == NULL) {
-        char *const q = OPENSSL_malloc(sz);
+        char *const q = OPENSSL_malloc(sz > 0 ? sz : 1);
 
         if (q == NULL)
             return 0;
         *val = q;
-        memcpy(q, p->data, sz);
+        if (sz == 0)
+            *q = '\0';
```

Comment:
This isn't necessary, as only octet strings may have zero length.  UTF8 string lengths should include a terminating NUL byte, so the "empty" UTF8 string is 1 byte, not 0.

Reply:
It was only intended as "don't pass uninitialized memory anywhere", though I could be convinced that if it stays inside a single provider it doesn't matter.